internal/pkg/mining/worker.go

```diff
@@ -194,46 +194,28 @@ func (w *DefaultWorker) Mine(ctx context.Context, base block.TipSet, nullBlkCoun
 		return
 	}
 
-	// lookback consensus.ElectionLookback
-	prevTicket, err := base.MinTicket()
-	if err != nil {
-		log.Warnf("Worker.Mine couldn't read parent ticket %s", err)
-		outCh <- Output{Err: err}
-		return
-	}
+	// Look-back for the election ticket.
+	// The parameter is interpreted as: lookback=1 means parent tipset. Subtract one here because the base from
+	// which the lookback is counted is already the parent, rather than "current" tipset.
+	// The sampling code will handle this underflowing past the genesis.
+	targetEpoch := baseEpoch - (miner.ElectionLookback - 1) + abi.ChainEpoch(nullBlkCount)
 
-	nextTicket, err := w.ticketGen.NextTicket(prevTicket, workerAddr, w.workerSigner)
+	nextTicket, err := w.ticketGen.MakeTicket(ctx, base.Key(), targetEpoch, w.minerAddr, workerAddr, w.workerSigner)
 	if err != nil {
 		log.Warnf("Worker.Mine couldn't generate next ticket %s", err)
 		outCh <- Output{Err: err}
 		return
 	}
-	// lookback consensus.ElectionLookback for the election ticket
-	baseHeight, err := base.Height()
-	if err != nil {
-		log.Warnf("Worker.Mine couldn't read base height %s", err)
-		outCh <- Output{Err: err}
-		return
-	}
-	ancestors, err := w.getAncestors(ctx, base, baseHeight+(abi.ChainEpoch(nullBlkCount+1)))
-	if err != nil {
-		log.Warnf("Worker.Mine couldn't get ancestorst %s", err)
-		outCh <- Output{Err: err}
-		return
-	}
-	electionTicket, err := sampling.SampleNthTicket(int(miner.ElectionLookback-1), ancestors)
-	if err != nil {
-		log.Warnf("Worker.Mine couldn't read parent ticket %s", err)
-		outCh <- Output{Err: err}
-		return
-	}
-	postRandomness, err := w.election.GeneratePoStRandomness(electionTicket, workerAddr, w.workerSigner, nullBlkCount)
+
+	postVrfProof, err := w.election.GenerateEPoStVrfProof(ctx, base.Key(), targetEpoch, w.minerAddr, workerAddr, w.workerSigner)
 	if err != nil {
-		log.Errorf("Worker.Mine failed to generate post randomness %s", err)
+		log.Errorf("Worker.Mine failed to generate epost postVrfProof %s", err)
 		outCh <- Output{Err: err}
 		return
 	}
-	powerTable, err := w.getPowerTable(ctx, base.Key())
+	postVrfProofDigest := blake2b.Sum256(postVrfProof)
```

Comment:
nit: we should use either vrfpi digest or the crypto make randomness seed function